### PR TITLE
Add initial CI for Arm64

### PR DESCRIPTION
**Pull request by eukarpov**

### CODE REVIEW

File: .github/workflows/ci-arm64.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+name: Build OpenSSL Arm64
+
+on:
+  pull_request:
+    branches: [ "master" ]
+  workflow_dispatch:
+    inputs:
+      openssl_build:
+        description: 'OpenSSL Arm64'
```
Comment (Blackhex) [inline .github/workflows/ci-arm64.yml]: I'd suggest to keep inputs description consistent and state explicitly that this build the OpenSSL with MSVC.

File: .github/workflows/ci-arm64.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    timeout-minutes: 600
+
+    steps:
+    
+    - name: Build OpenSSL Arm64
+      shell: cmd
+      run: |
+        echo on
+        rmdir /s /q openssl
+        git clone https://github.com/openssl/openssl.git --single-branch --depth 1
```
Comment (Blackhex) [inline .github/workflows/ci-arm64.yml]: Is there any specific reason why the standard [https://github.com/actions/checkout](https://github.com/actions/checkout) is not used as a separate step?

File: .github/workflows/ci-arm64.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        cd openssl
+        call "%ProgramFiles%\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvarsall.bat" arm64
```
Comment (Blackhex) [inline .github/workflows/ci-arm64.yml:32]: I'd suggest moving the detailed steps for building into a separate batch script for concerns separation and better maintenance.

File: .github/workflows/ci-arm64.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        set VSCMD_DEBUG=3
+        set PATH=%PATH%;%VCINSTALLDIR%\Tools\Llvm\ARM64\bin
+        perl Configure VC-CLANG-WIN64-CLANGASM-ARM
+        nmake
+        nmake test
+        .\apps\openssl.exe speed 2> benchmark.txt
+        type benchmark.txt
+
+  build-openssl-aarch64:
```
Comment (Blackhex) [inline .github/workflows/ci-arm64.yml]: It's not straightforward from the job and steps name that this is a Linux build vs. the previous Windows builds. I'd suggest to state "Windows" and "Linux" in that jobs explicitly.

**Review by Blackhex — CHANGES_REQUESTED** [review]
Mostly code clarity related comments, otherwise it looks good.

File: .github/workflows/ci-arm64.yml (hunk 1 above)
Comment (pragmanomos) [reply .github/workflows/ci-arm64.yml]: I would go further and explicitly specify, for each build, OS, architecture and toolchain.

File: .github/workflows/ci-arm64.yml (hunk 2 above)
Comment (eukarpov) [reply .github/workflows/ci-arm64.yml]: Not really, it has been changed

File: .github/workflows/ci-arm64.yml (hunk 4 above)
Comment (eukarpov) [reply .github/workflows/ci-arm64.yml]: Changed

File: .github/workflows/ci-arm64.yml (hunk 1 above)
Comment (eukarpov) [reply .github/workflows/ci-arm64.yml]: Changed

**Review by pragmanomos — APPROVED** [review]

File: .github/workflows/ci-arm64.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      run: |
+        echo on
+        call "%ProgramFiles%\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvarsall.bat" arm64
+        set VSCMD_DEBUG=3
+        set PATH=%PATH%;%VCINSTALLDIR%\Tools\Llvm\ARM64\bin
+        perl Configure VC-CLANG-WIN64-CLANGASM-ARM
+        nmake
+        nmake test
+        .\apps\openssl.exe speed 2> benchmark.txt
+        type benchmark.txt
```
Comment (vladimir-aubrecht) [inline .github/workflows/ci-arm64.yml:60]: Shouldn't this be ``cat benchmark.txt``?

File: .github/workflows/ci-arm64.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    - name: Git checkout
+      uses: actions/checkout@v3
+    
+    - name: Build OpenSSL on Windows Arm64 with MSVC
+      shell: cmd
+      run: |
+        echo on
+        call "%ProgramFiles%\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvarsall.bat" arm64
+        set VSCMD_DEBUG=3
+        set PATH=%PATH%;%VCINSTALLDIR%\Tools\Llvm\ARM64\bin
```
Comment (Blackhex) [inline .github/workflows/ci-arm64.yml:34]: Why is adding LLVM to the path needed for the MSVC build?

File: .github/workflows/ci-arm64.yml (hunk 5 above)
Comment (Blackhex) [reply .github/workflows/ci-arm64.yml:60]: There is no cat in Batch.

File: .github/workflows/ci-arm64.yml (hunk 6 above)
Comment (eukarpov) [reply .github/workflows/ci-arm64.yml:34]: a good suggestion, I will think about it for the next iteration.

**Pull request merged**